packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -629,20 +636,24 @@ class IconButton extends StatelessWidget {
     InteractiveInkFeatureFactory? splashFactory,
   }) {
     final MaterialStateProperty<Color?>? buttonBackgroundColor = (backgroundColor == null && disabledBackgroundColor == null)
-        ? null
-        : _IconButtonDefaultBackground(backgroundColor, disabledBackgroundColor);
+      ? null
+      : _IconButtonDefaultBackground(backgroundColor, disabledBackgroundColor);
     final MaterialStateProperty<Color?>? buttonForegroundColor = (foregroundColor == null && disabledForegroundColor == null)
+      ? null
+      : _IconButtonDefaultForeground(foregroundColor, disabledForegroundColor);
+    final MaterialStateProperty<Color?>? overlayColorProp = (foregroundColor == null &&
+      hoverColor == null && focusColor == null && highlightColor == null && overlayColor == null)
         ? null
-        : _IconButtonDefaultForeground(foregroundColor, disabledForegroundColor);
-    final MaterialStateProperty<Color?>? overlayColor = (foregroundColor == null && hoverColor == null && focusColor == null && highlightColor == null)
-        ? null
-        : _IconButtonDefaultOverlay(foregroundColor, focusColor, hoverColor, highlightColor);
+        : switch (overlayColor) {
```

Comment:
This switch statement is pretty slick. Are the parens in the else clause - `(_)` needed?

Reply:
Good question, i checked other resources, They're not using parens. Fixed.